msgraph-oauth-feature/src/com/microsoft/auth/OAuth2Feature.java

```diff
@@ -127,6 +127,16 @@ else  if (isUserPassAuth(config)) {
     return form;
   }
 
+  private static String grantType(FeatureConfig config) {
```

Comment:
so here we see the three kinds of authentication, right?

- isAppAuth: client_credentials:  App Based Login with tenantId, clientId, clientSecret
- isUserPassAuth: password: The user-password auth flow used for technical users, because some API's need a "user flow"
- authorization_code: The regular flow based on a user session


-